### ISSUE TITLE
Do we have the right daylight savings days?

### ISSUE THREAD

**Issue by rmkaplan** [Interlisp/medley#544]
\UNPACKDATE seems to return an hour that is one less than our current DST clock time.  But its return value-component to say that it is DST is NIL, at least today.  If you give it a date from a week or so ago, it gets it right, or a week have Novermber 7.

I think it doesn't have the right settings for the begin/end of DST.  Constants in IOCHAR?

**Comment by D-Van-Buer** [comment]
TCP/IP NTP time protocol does not deal with time zones or DST at all, it just establishes time in seconds (and fraction) from 1970.
In the Linux runtime code, there is a fairly large (and probably messy) table that isn't just the days it changes now, but for all years so that the time format transformation calls do the right thing for past dates too.  This means there are updates to the Linux code tree that get pushed out every time the DST laws change somewhere.  Conversions in time format are handled in the C libraries.
I think it would be possible to write code that could deduce the magic numbers.
I don't think the Interlisp user base will be large enough anytime soon to be worth doing the "right thing".

**Comment by nbriggs** [comment]
Yeah, absolutely, which is why the values are trivially settable by anyone who wants to put them in a site or personal init file, without recompilation or anything magic… ;-)

**Comment by masinter** [comment]
Belongs in the FAQ and intro to running Medley... especially a problem for online users since the user could be coming from anywhere.   There are filedate problems for FILECREATED dates because DATE and IDATE don't do anything with TimeZone info .

**Comment by orcmid** [comment]
It's probably safe to use a universal time (whether seconds from an origin), as long as the origin timezone is understood.  I assume that is UTC in modern systems.  Some operating systems don't handle that with regard to showing local time properly and/or stamping local files properly.  With regard to persistent dates of a "now" somewhere, it should be "now" where the date is affixed.  If that is always UTC, then it is easy for an observer to convert to their relevant local time.  That seems to work on email that I receive from international origins.

I think it is amazing that Linux handles retrospective daylight rules.  It is annoying for me that my Outlook Calendar makes a mess of it when I look back at older entries.

With regard to Interlisp adoption, look at it this way.  The user base can't get internationally (or even nationally) larger if this is not handled.  And whatever course is taken, it must be sensible to some degree of sensibility.  

As I type this, there are 43,457 of over a million users active in the Second Life virtual world.  The local "SL Time" is 9:39 AM.  That happens to be USA California local time and everyone copes with that.  All in-game times are recorded in SL Time, including on timestamps of messages.  I don't mind.  I live in the same time zone :).   The fixed, predictable nature of SL Time is something all can deal with, including the contrary (to the Southern Hemisphere) or offset differences in daylight times, where those occur.

And I adjust my head around the time of distant connections the same as I would in arranging or participating in conference calls and web casts, the last two being in Haifa and in Melbourne, respectively.
